router/broker.go

```diff
@@ -305,47 +365,80 @@ func (b *broker) syncPublish(pub *wamp.Session, msg *wamp.Publish, pubID wamp.ID
 }
 
 func newSubscription(id wamp.ID, subscriber *wamp.Session, topic wamp.URI, match string) *subscription {
+	subscribers := map[*wamp.Session]struct{}{}
+	if subscriber != nil {
+		subscribers[subscriber] = struct{}{}
+	}
+
 	return &subscription{
 		id:          id,
 		topic:       topic,
 		match:       match,
 		created:     wamp.NowISO8601(),
-		subscribers: map[*wamp.Session]struct{}{subscriber: {}},
+		subscribers: subscribers,
 	}
 }
 
-func (b *broker) syncSubscribe(subscriber *wamp.Session, msg *wamp.Subscribe, match string) {
-	var sub *subscription
-	var existingSub bool
+func (b *broker) syncSaveEvent(eventStore *historyStore, pub *wamp.Publish, event *wamp.Event) {
+
+	if eventStore.isLimitReached {
```

Comment:
Given that len of a slice is free to compute, I would avoid the extra state and just use the length test below.

Reply:
Well, `isLimitReached` is anyway there as it can be useful from the business view. So not much changed use it of length in the condition.